sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotHardcodeCredentials.cs

```diff
@@ -33,39 +33,46 @@ namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     [Rule(DiagnosticId)]
-    public sealed class DoNotHardcodeCredentials : SonarDiagnosticAnalyzer
+    public sealed class DoNotHardcodeCredentials : ParameterLoadingDiagnosticAnalyzer
     {
         internal const string DiagnosticId = "S2068";
         private const string MessageFormat = "Remove hard-coded password(s): '{0}'.";
 
         private static readonly DiagnosticDescriptor rule =
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+
+        private IEnumerable<string> splitCredentialWords;
+        private string credentialWords;
+        private Regex passwordValuePattern;
+
+        private const string DefaultCredentialWords = "password, passwd, pwd";
+        [RuleParameter("credentialWords", PropertyType.String, "Comma separated list of words identifying potential credentials",
+            DefaultCredentialWords)]
+        public string CredentialWords
+        {
+            get { return credentialWords; }
+            set
+            {
+                credentialWords = value;
+                splitCredentialWords = value.ToLowerInvariant().Split(',').Select(x => x.Trim()).ToList();

```

Comment:
Need to add `StringSplitOptions.RemoveEmptyEntries`, otherwise regex would be wrong for `foo,bar,` or `foo,,bar`. Please also add tests for that.

Reply:
Wasn't aware of this option + good catch for multiple ','.